File: pascal/division.pas
program DivisiondeDeuxnombres;

var
  nombre1, nombre2, resultat: real;

begin
  (* Demandez à l'utilisateur de saisir deux nombres *)
  writeln('Entrez le premier nombre : ');
  readln(nombre1);

  writeln('Entrez le second nombre : ');
  readln(nombre2);

  (* verifiez si le second nombre est zéro pour eviter la division par zéro *)
  if nombre2 = 0 then
  writeln('Erreur : division par zéro non permise.')
else
begin
  (* Calculez la division des deux nombres *)
  resultat:= nombre1 / nombre2;

  (* Affichez le resultat *)
  writeln('le resultat de ', nombre1:0:2, ' divisé par ', nombre2:0:2, ' est : ', resultat:0:2)

end;
end.

File: python/division.py
def division(a, b):
    return a / b

N1 = float(input("Entrer le premier nombre : "))    
N2 = float(input("Entrer le second nombre : "))

if N1 == 0 :
    print(f"La division de {N1} par {N2} est 0")
elif N2 == 0 :
    print(f"Il est impossible de diviser {N1} par {N2}")
else :
    S = division(N1, N2)
    print(f"La division de {N1} par {N2} est {S:.2f}")
        
